src/components/AddItemForm.js

```diff
@@ -1,28 +1,60 @@
-import React, {useState} from 'react';
+import React, {useState, useEffect} from 'react';
 import fb from '../lib/firebase';
 import '../css/AddItemForm.css';
-import { v4 as uuidv4 } from 'uuid';
+// import { Redirect } from 'react-router-dom';
 
 const Form = ({token}) => {
     const [itemName, setItemName] = useState("");
     const [timeFrame, setTimeFrame] = useState(7);
     const [lastPurchaseDate, setPurchaseDate] = useState(null);
     const userToken = token || "userToken";
+    const [shoppingListCollection, setShoppingListCollection] = useState([]);
+    const [duplicateError, setDuplicateError] = useState(false);
+
+    useEffect(() => {
+        const db =  fb.firestore()
+        const tokenRef = db.collection(token);
+
+        tokenRef
+            .orderBy("timeFrame", "asc")
+            .get()
+            .then((querySnapshot) => {
+                let fullCollection = [];
+
+                querySnapshot.forEach((doc) => {
+                    let documentData = doc.data();
+                    let nameData = documentData.itemName
+                    nameData = nameData.toLowerCase().replace(/[.,\/#!$%\^&\*;:{}=\-_`~()]/g,"").trim().replace(/\s{2,}/g," ");
+
+                    fullCollection.push(nameData);
+                });
+                setShoppingListCollection(fullCollection);
+            }).catch((error) => {
+                console.log("Error getting document:", error);
+            });
+    }, []);
 
     const handleSubmit = e => {
         e.preventDefault();
         let db = fb.firestore();
-        let data = {
-            id: uuidv4(),
-            itemName,
-            timeFrame: parseInt(timeFrame),
-            lastPurchaseDate
-        };
-        db.collection(userToken).add(data)
-        .then(() => alert(" successfully written!"))
-        .catch(error => console.error("Error writing document: ", error));
+        let tokenRef = db.collection(userToken)
+        let normalizeItemName = itemName.toLowerCase().replace(/[.,\/#!$%\^&\*;:{}=\-_`~()]/g,"").trim().replace(/\s{2,}/g," ");
+        if (!shoppingListCollection.includes(normalizeItemName)) {
+            let data = {
+                itemName,
+                timeFrame: parseInt(timeFrame),
+                lastPurchaseDate
+            };
+            tokenRef.add(data)
+            .then((docRef) => { tokenRef.doc(docRef.id).update({ id : docRef.id });  })
```

Comment:
> Looks good. Just so I'm understanding, we are using the id given by Firestore instead of implementing uuid? Is that what I'm seeing? Otherwise, looks good as always.

That is correct. If you take a look at this line, the add data promise to Firebase returns a docRef response. We then take the ID portion of that response and assign it to our document, eliminating the unnecessary use of the uuid package.

Reply:
```suggestion
            .then((docRef) => { tokenRef.doc(docRef.id).update({ id : docRef.id });  setShoppingListCollection(shoppingListCollection.concat(data))})
```
This will trigger the useEffect above so that you have the latest shopping list items from the db to compare.

Maybe at the beginning of the `handleSubmit` we could `setDuplicateError(false)` so that the error goes away when first submitting an item. It can be confusing because the error is always there once it's triggered.

Reply:
If the new data is joined with the shoppingListCollection to trigger useEffect, won't shoppingListCollection be overwritten with the Firebase call in useEffect?

Reply:
I think so, yes. So I'm thinking of this situation:

1. List loads with apples, bananas, garlic
2. I successfully add garbage bags to the list via the add item form.
3. Right now if I try to enter garbage bags again, it wouldn't see it is a duplicate, because react state doesn't know that garbage bags is in the list (it only knows about apples, bananas and garlic). 
4. You can do it in a different way, but after adding an item to the list, you want to make sure your local state has that item in the list, too. Otherwise, you'll be able to add that item over and over again.

Reply:
That makes sense. I was thinking that it would redirect back to /AddItem on Submit, but we have a prevent default up there at the top of handleSubmit.